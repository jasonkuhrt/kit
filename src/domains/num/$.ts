/**
 * Numeric types and utilities with branded types for mathematical constraints.
 *
 * Provides branded number types (Positive, Negative, Even, Odd, etc.) with
 * runtime validation, mathematical operations, range types, and specialized
 * numeric domains like Complex, Ratio, and BigInt. Includes type guards,
 * ordering, and equivalence utilities.
 */
export * as Num from './$$.js'
