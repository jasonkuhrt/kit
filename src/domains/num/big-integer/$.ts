export * as BigInteger from './$$.ts'
