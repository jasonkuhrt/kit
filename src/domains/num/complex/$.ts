export * as Complex from './$$.js'
