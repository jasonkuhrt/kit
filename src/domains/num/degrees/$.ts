// Export the namespace containing all Degrees operations
export * as Degrees from './$$.js'
