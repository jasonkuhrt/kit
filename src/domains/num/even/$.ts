// Export the namespace containing all Even operations
export * as Even from './$$.ts'
