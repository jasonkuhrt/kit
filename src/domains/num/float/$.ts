// Export the namespace containing all Float operations
export * as Float from './$$.js'
