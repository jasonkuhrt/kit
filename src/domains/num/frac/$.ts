// Export the namespace containing all Frac operations
export * as Frac from './$$.js'
