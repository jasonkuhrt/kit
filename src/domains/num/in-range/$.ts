// Export the namespace containing all InRange operations
export * as InRange from './$$.js'
