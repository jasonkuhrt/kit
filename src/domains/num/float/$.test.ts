import { expect, test } from 'vitest'
import { Float } from './$.js'

test('isFloat', () => {
  // TODO: Add comprehensive tests
  expect(Float.is(0)).toBe(false)
})

test('float', () => {
  // TODO: Add comprehensive tests
  expect(() => Float.from(0)).toThrow()
})

test('tryFloat', () => {
  // TODO: Add comprehensive tests
  expect(Float.tryFrom(0)).toBe(null)
})
