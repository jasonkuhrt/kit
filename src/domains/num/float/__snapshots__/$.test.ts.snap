// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`from > from(-0.1) > from(-0.1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-0.1
╠══════════════════════════════════════════════════╣ THEN RETURNS
-0.1
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(-1) > from(-1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-1
╠══════════════════════════════════════════════════╣ THEN THROWS
Error: Value must not be an integer, got: -1
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(-1.5) > from(-1.5) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-1.5
╠══════════════════════════════════════════════════╣ THEN RETURNS
-1.5
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(0) > from(0) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN THROWS
Error: Value must not be an integer, got: 0
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(0.1) > from(0.1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0.1
╠══════════════════════════════════════════════════╣ THEN RETURNS
0.1
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(1) > from(1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1
╠══════════════════════════════════════════════════╣ THEN THROWS
Error: Value must not be an integer, got: 1
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(1.5) > from(1.5) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.5
╠══════════════════════════════════════════════════╣ THEN RETURNS
1.5
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(3.141592653589793) > from(3.141592653589793) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
3.141592653589793
╠══════════════════════════════════════════════════╣ THEN RETURNS
3.141592653589793
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(null) > from(null) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN THROWS
Error: Value must be finite, got: Infinity
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(null) > from(null) 2`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN THROWS
Error: Value must be finite, got: NaN
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is("1.5") > is("1.5") 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
"1.5"
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is() > is() 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS

╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(-0.1) > is(-0.1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-0.1
╠══════════════════════════════════════════════════╣ THEN RETURNS
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(-1) > is(-1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-1
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(-1.5) > is(-1.5) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-1.5
╠══════════════════════════════════════════════════╣ THEN RETURNS
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(0) > is(0) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(0.1) > is(0.1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0.1
╠══════════════════════════════════════════════════╣ THEN RETURNS
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(1) > is(1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(1.5) > is(1.5) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.5
╠══════════════════════════════════════════════════╣ THEN RETURNS
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(3.141592653589793) > is(3.141592653589793) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
3.141592653589793
╠══════════════════════════════════════════════════╣ THEN RETURNS
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(100) > is(100) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
100
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) > is(null) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) > is(null) 2`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) > is(null) 3`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) > is(null) 4`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(-0.1) > tryFrom(-0.1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-0.1
╠══════════════════════════════════════════════════╣ THEN RETURNS
-0.1
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(-1) > tryFrom(-1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-1
╠══════════════════════════════════════════════════╣ THEN RETURNS
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(-1.5) > tryFrom(-1.5) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-1.5
╠══════════════════════════════════════════════════╣ THEN RETURNS
-1.5
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(0) > tryFrom(0) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN RETURNS
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(0.1) > tryFrom(0.1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0.1
╠══════════════════════════════════════════════════╣ THEN RETURNS
0.1
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(1) > tryFrom(1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1
╠══════════════════════════════════════════════════╣ THEN RETURNS
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(1.5) > tryFrom(1.5) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.5
╠══════════════════════════════════════════════════╣ THEN RETURNS
1.5
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(3.141592653589793) > tryFrom(3.141592653589793) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
3.141592653589793
╠══════════════════════════════════════════════════╣ THEN RETURNS
3.141592653589793
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(null) > tryFrom(null) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(null) > tryFrom(null) 2`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
null
╚══════════════════════════════════════════════════╝"
`;
